internal/refreshtoken/partial_resource.go

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package refreshtoken
+
+import (
+	"time"
+
+	"github.com/hashicorp/boundary/internal/db/timestamp"
+	"github.com/hashicorp/boundary/internal/types/resource"
+)
+
+type partialResource struct {
+	publicId     string
+	updateTime   time.Time
+	resourceType resource.Type
+}
```

Comment:
This stands out to me as something odd. Partly due to the name "partial resource". While this is not exported, the same name is used elsewhere as an interface that this happens to implement. In fact there are several interfaces, one per domain package that are all the same, and this happens to implement them all. 

I think this might be cleaned up an simplified by a type in the `pagination` package.

```go
package pagination

type Item struct {
  PublicId   string
  UpdateTime time.Time
  Type       resource.Type
}
```

```go
package refreshtoken

func (rt *Token) LastItem() *pagination.Item {
  // alternatively, just change Token
  // to have a LastItem field of type pagination.Item
  return &pagination.Item{
    PublicId: rt.LastItemId,
    UpdateTime: rt.LastItemUpdatedTime,
    Type: rt.ResourceType,
  }
}
```

```go
package target

func WithStartPageAfterItem(item *pagination.Item) Option {
  return func(o *options) {
    o.WithStartPageAfterItem = item
  }
}
```

Reply:
Although looking at https://github.com/hashicorp/boundary/pull/3844 that is updating the `pagination` package to import `refreshtoken`, so the suggestion above would create an import cycle. So maybe `Item` should be in `refreshtoken`.

Reply:
This doesn't change the need for an interface is the domain packages, since we need to support both using a `boundary.Resource` and an item from a refresh token. See the changes to `service_list_refresh.go` here: https://github.com/hashicorp/boundary/pull/3840/commits/8a20fb8f0c9221d64a91be13490cb3a5ca247785#diff-c8bedb31d2abcb15d2ea66a841d4a2608aa7bc879a089ad6a8c077ea661d9545L68. I'm happy to move the type around, but I still think we need the interface in each domain.

Reply:
Maybe have the interface in `pagination` then? ie:

```go
package pagination

type Item interface {
  GetPublicId() string
  GetUpdateTime() time.Time
  GetResourceType() resource.Type
}
```

```go
package refreshtoken

type Item struct {} // implements pagination.Item

func (rt *Token) LastItem() *Item {
  // alternatively, just change Token
  // to have a LastItem field of type Item
  return &Item{
    PublicId: rt.LastItemId,
    UpdateTime: rt.LastItemUpdatedTime,
    Type: rt.ResourceType,
  }
}
```

```go
package target

func WithStartPageAfterItem(item pagination.Item) Option {
  return func(o *options) {
    o.WithStartPageAfterItem = item
  }
}
```

Reply:
That works for me!